framework/test/jest/specs/unit/controller/channels/child_process_channel.spec.js

```diff
@@ -0,0 +1,622 @@
+	const params = {
+		moduleAlias: originalModuleAlias,
+		events: ['event1', 'event2'],
+		actions: {
+			action1: jest.fn(),
+			action2: jest.fn(),
+			action3: jest.fn(),
+		},
+		options: {},
+	};
+	let childProcessChannel;
+	const socketsPath = {
+		root: 'root',
+		sub: 'sub',
+		pub: 'pub',
+		rpc: 'rpc',
+	};
+
+	let resolveWhenAllSocketsBound;
+	let rejectWhenAnySocketFailsToBind;
+	let rejectWhenTimeout;
+	let removeAllListeners;
+
+	beforeEach(() => {
+		axon.socket = jest.fn().mockReturnValue({
+			connect: jest.fn(),
+			close: jest.fn(),
+			on: jest.fn(),
+			once: jest.fn((event, callback) => {
+				callback();
+			}),
+			bind: jest.fn(),
+			emit: jest.fn(),
+			removeAllListeners: jest.fn(),
+			sock: {
+				once: jest.fn((event, callback) => {
+					callback();
+				}),
+				on: jest.fn((event, callback) => {
+					callback();
+				}),
+				removeAllListeners: jest.fn(),
+			},
+		});
+
+		childProcessChannel = new ChildProcessChannel(
+			params.moduleAlias,
+			params.events,
+			params.actions,
+			params.options
+		);
+
+		resolveWhenAllSocketsBound =
+			childProcessChannel._resolveWhenAllSocketsBound;
+		rejectWhenAnySocketFailsToBind =
+			childProcessChannel._rejectWhenAnySocketFailsToBind;
+		rejectWhenTimeout = childProcessChannel._rejectWhenTimeout;
+		removeAllListeners = childProcessChannel._removeAllListeners;
+
+		childProcessChannel._resolveWhenAllSocketsBound = jest.fn();
+		childProcessChannel._rejectWhenAnySocketFailsToBind = jest.fn();
+		childProcessChannel._rejectWhenTimeout = jest.fn();
+		childProcessChannel._removeAllListeners = jest.fn();
+	});
+
+	describe('inheritance', () => {
+		it('should be extended from BaseChannel class', () => {
+			// Assert
+			expect(Object.getPrototypeOf(childProcessChannel)).toBeInstanceOf(
+				BaseChannel
+			);
+		});
+
+		it('should call BaseChannel class constructor with arguments', () => {
+			// Arrange
+			let IsolatedChildProcessChannel = null;
+			let IsolatedBaseChannel = null;
+
+			jest.isolateModules(() => {
+				jest.doMock('../../../../../../src/controller/channels/base_channel');
+				IsolatedChildProcessChannel = require('../../../../../../src/controller/channels/child_process_channel');
+				IsolatedBaseChannel = require('../../../../../../src/controller/channels/base_channel');
+			});
+
+			// Act
+			childProcessChannel = new IsolatedChildProcessChannel(
+				params.moduleAlias,
+				params.events,
+				params.actions,
+				params.options
+			);
+
+			// Assert
+			expect(IsolatedBaseChannel).toHaveBeenCalledWith(
+				params.moduleAlias,
+				params.events,
+				params.actions,
+				params.options
+			);
+		});
+	});
+
+	describe('#constructor', () => {
+		it('should create a local bus based on EventEmitter2', () => {
+			// Assert
+			expect(childProcessChannel.localBus).toBeInstanceOf(EventEmitter2);
+		});
+	});
+
+	describe('#registerToBus', () => {
+		beforeEach(() => childProcessChannel.registerToBus(socketsPath));
```

Comment:
prefer to consistently use 2 lines for test
```
beforeEach(async () => {
  childProcessChannel.registerToBus(socketsPath)
});
```

Reply:
This has conflicts with `eslint` rule with `arrow-body-style`. I don't see the problem when there is only one line.

Reply:
mmm i c, then it's ok.
In elements, we had this convention to always not to use one-liner, but i guess jest setup is different